### PR TITLE
revert to Jackson BigInteger and BigDecimal serialization

### LINKED ISSUE #8387: Remove custom serializers/de-serializers for bigint and bignum
As introduced in https://github.com/elastic/logstash/pull/8205, it appears that this may be superseded by a newer version of Jackson that properly supports those types, as evident by this comment : https://github.com/elastic/logstash/issues/8379#issuecomment-332010164. 

Jackson 2.9.1 is introduced with PR https://github.com/elastic/logstash/pull/8373

I _think_ we can remove these custom serialization. 

cc/ @original-brownbear

### PR DESCRIPTION
**Pull request by colinsurprenant**
Revert the custom serializers that are not required anymore with Jackson version 2.8+ now that #7615 #8373  is in. 

Relates to #8379, #8205, #8131

Fixes #8387

### CODE REVIEW

**Comment by colinsurprenant** [comment]
ping @original-brownbear @guyboertje @jordansissel

**Comment by jordansissel** [comment]
LGTM

**Comment by original-brownbear** [comment]
@jordansissel @colinsurprenant this will have somewhat negative performance repercussions, especially for the Java execution but also in the current Ruby execution.
The reason is that we're now serializing `RubyBignum` to Java's `BigInteger` when going through the full roundtrip => every `get` on the `RubyBignum` field will now entail one instantiation of `RubyBignum`.

Can't we simply move the serialize as `String` to using `.writeNumber ` as we did here and also move the deserialization to reading `Number`? That would be even faster than the current solution and also wouldn't force another adjustment of the Java execution :)

**Comment by original-brownbear** [comment]
@colinsurprenant @jordansissel that said, in principle, I'm really in favor of serializing (and handling this conversion) exactly the way it's done here.
But at least @andrewvc insisted in the Java execution PR, that the Ruby behavior for comparisons and such in filters must be retained `100%`, as long as that's the case, this (disregarding the Ruby type and forcing everything to Java types after deserialization) is a big performance (or complexity in upstream code) hit :(

**Comment by colinsurprenant** [comment]
@original-brownbear I am not sure I understand the impact in the Java execution and the «big performance (or complexity in upstream code) hit»? Do you mean impact and/or performance induced by this  PR? Could you expand on this please?

Also, unless I am misunderstanding you, compatibility with current plugins is not affected by this PR. 

Before this PR, for an `Event` with a `RubyBignum` field value:
- `RubyBignum` is converted to a String for serialization
- Deserialization re-instantiate a RubyBigNum from the serialized string

And hhis PR changes this to:
- `RubyBignum` is converted to `BigInteger` and serialized by Jackson internal CBOR serializer 
- Deserialization re-instantiate a `BigInteger`using  Jackson internal CBOR deserializer and is placed into the Event as-is.

So Basically the *only* differences are:
1- this PR avoids and explicit `RubyBignum` `.toString()` conversion on serialization
2- this PR injects a `BigInteger` into the Event instead of explicitly recreating a `RubyBignum` by parsing its string representation. 

Are you only suggesting we reinject a `RubyBignum` from the `BigInteger` upon deserialization? It would probably make sense from a Ruby performance perspective where the `RubyBignum` would exist once and the `BigInteger` value could be extracted and handed off to the Java API. But similarly we don't do that for `boolean` or `double` scalar values which are stored in the event as is so each Ruby get on these fields instantiate a new `RubyBoolean` for example.

**Comment by original-brownbear** [comment]
@colinsurprenant 

> each Ruby get on these fields instantiate a new RubyBoolean for example.

`boolean` is a special case, there are no instantiations there in the converter since `TRUE` and `FALSE` are singletons.

`double` and `int` def are wrong, grml. Haven't seen those before :) Actually, in that case, feel free to go ahead here. No point in having an optimized path for these super rare cases and not have it for `double` and `int`.

-> LGTM :)

**Comment by colinsurprenant** [comment]
@original-brownbear Great! But I am not against the idea of actually wrapping the deserialized Java `BigInteger` and `BigDecimal` into their Ruby equivalent either since we *can't loose* given that the underlying Java value is preserved in the Ruby object so it's a win-win on both sides. I guess `boolean` is an exception because of the singleton situation but we can certainly also optimize `int` and `double` in the same way, WDYT? 

I'll go ahead and merge this and then we can submit another PR to optimize?

**Comment by original-brownbear** [comment]
@colinsurprenant jup, my thoughts exactly :) We can optimize this down the line in some other PR. Makes things much easier to read got log wise if we do all the numbers in one go anyhow.

**Comment by colinsurprenant** [comment]
@original-brownbear great work on the BiValue refactor BTW! love it!

**Comment by colinsurprenant** [comment]
merged in master and 6.x.
